lib/routes/sign.js

```diff
@@ -88,23 +88,38 @@ module.exports = function (log, isA, error, signer, db, domain) {
           }
         }
         var uid = sessionToken.uid.toString('hex')
-        signer.sign(
-          {
-            email: uid + '@' + domain,
-            publicKey: publicKey,
-            domain: domain,
-            duration: duration,
-            generation: sessionToken.verifierSetAt,
-            lastAuthAt: sessionToken.lastAuthAt(),
-            verifiedEmail: sessionToken.email
-          }
-        ).then(function(certResult) {
-          log.activityEvent('account.signed', request, {
-            uid: uid,
-            account_created_at: sessionToken.accountCreatedAt
-          })
-          reply(certResult)
-        }, reply)
+
+        db.sessionDevice(sessionToken.tokenId)
```

Comment:
I'm not sure. I didn't want to impose the extra work on every sessionToken fetch but it might be worth comparing the two approaches. The benefit here may outweigh the downside elsewhere.